frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx

```diff
@@ -47,9 +58,70 @@ const MapSideBarContainer: React.FunctionComponent<IMapSideBarContainerProps> =
     state => state.parcel?.parcelDetail as IPropertyDetail,
   );
   const [cachedParcelDetail, setCachedParcelDetail] = React.useState<IParcel | null>(null);
+
+  const [movingPinNameSpace, setMovingPinNameSpace] = useState<string | undefined>(
+    movingPinNameSpaceProp,
+  );
   const leafletMouseEvent = useSelector<RootState, LeafletMouseEvent | null>(
     state => state.leafletClickEvent?.mapClickEvent,
   );
+  const [propertyType, setPropertyType] = useState('');
+  const formikRef = React.useRef<FormikValues>();
+  const { handleGeocoderChanges } = useGeocoder({ formikRef });
+  const parcelsService = useLayerQuery(PARCELS_LAYER_URL);
+
+  const handlePidChange = (pid: string) => {
+    const formattedPid = pid.replace(/-/g, '');
+    const response = parcelsService.findByPid(formattedPid);
+    handleParcelDataLayerResponse(response, dispatch);
+  };
+  const handlePinChange = (pin: string) => {
+    const response = parcelsService.findByPin(pin);
+    handleParcelDataLayerResponse(response, dispatch);
+  };
+
+  React.useEffect(() => {
+    if (movingPinNameSpace !== undefined) {
+      document.body.className = propertyType === 'building' ? 'building-cursor' : 'parcel-cursor';
+    }
+    return () => {
+      //make sure to reset the cursor when this component is disposed.
+      document.body.className = '';
+    };
+  }, [propertyType, movingPinNameSpace]);
+
+  //Add a pin to the map where the user has clicked.
+  useDeepCompareEffect(() => {
+    //If we click on the map, create a new pin at the click location.
+    if (
+      movingPinNameSpace !== undefined &&
+      !!formikRef?.current &&
+      isMouseEventRecent(leafletMouseEvent?.originalEvent)
+    ) {
+      let nameSpace = (movingPinNameSpace?.length ?? 0) > 0 ? `${movingPinNameSpace}.` : '';
+      if (propertyType === 'land') {
+        formikRef.current.setFieldValue(
+          `${nameSpace}data.latitude`,
+          leafletMouseEvent?.latlng.lat || 0,
+        );
+        formikRef.current.setFieldValue(
+          `${nameSpace}data.longitude`,
+          leafletMouseEvent?.latlng.lng || 0,
+        );
+      } else {
+        formikRef.current.setFieldValue(
+          `data.buildings.0.latitude`,
```

Comment:
will have to update how these are handled with next iteration of wizard with multiple buildings